rskj-core/src/main/java/co/rsk/net/sync/SyncMessageHandler.java

```diff
@@ -64,12 +69,24 @@ public void run() {
 
         Job job = null;
         while (isRunning()) {
+            Metric metric = null;
             try {
                 job = jobQueue.take();
 
+                metric = profiler.start(job.getMetricKind());
+
+                if (maxSenderJobs != null) {
+                    Peer sender = job.getSender();
+                    NodeID senderId = sender.getPeerNodeID();
+                    long jobCount = 1 + jobQueue.stream().filter(j -> j.getSender().getPeerNodeID().equals(senderId)).count();
```

Comment:
Wondering if there is a way to keep the count in a map by senderId so we avoid filtering + counting in the loop. Also not sure how many jobs could be there. If it is possible to have many jobs then that might be a problem. 

Do you think it is worth to think about adding a map where the key is the sender id and value is the count for this purpose?

Reply:
Souldn't that logic only work for snap?